core/src/autogluon/core/scheduler/scheduler.py

```diff
@@ -130,6 +140,24 @@ def _release_resource_callback(fut):
         job.add_done_callback(_release_resource_callback)
         return job
 
+    @staticmethod
+    def _worker(pickled_fn, return_list, gpu_ids, args):
+        """Worker function in thec client
+        """
+        fn = dill.loads(pickled_fn)
+        if len(gpu_ids) > 0:
+            # handle GPU devices
+            os.environ['CUDA_VISIBLE_DEVICES'] = ",".join(map(str, gpu_ids))
+            os.environ['MXNET_CUDNN_AUTOTUNE_DEFAULT'] = "0"
```

Comment:
I find that we will need to add the following to correctly disabling GPU when `ngpu_per_trial` is set to 0.
```
if len(gpu_ids) > 0:
    # handle GPU devices
    os.environ['CUDA_VISIBLE_DEVICES'] = ",".join(map(str, gpu_ids))
    os.environ['MXNET_CUDNN_AUTOTUNE_DEFAULT'] = "0"
else:
    os.environ['CUDA_VISIBLE_DEVICES'] = "-1"
```

Reply:
Added

Reply:
On the subject above:

> I know that this is WIP but I'd like to provide the example to help us locate potential issues. I tried the following code with the changes in this PR. But it reports an error:
...
> AttributeError: Can't get attribute 'Net' on <module '__main__' (built-in)>

Every object you want to pass into HPO must be serializable 'on-the-other-side'. Here is how can you re-frame the task:
```python
import numpy as np
import autogluon.core as ag
import torch as th
import torch.nn as nn
import dill

@ag.args(
    num_hidden=ag.space.Int(16, 32),
    lr=ag.space.Real(1e-3, 1e-2)    
)
def train_nn(args, reporter):
    num_hidden = args['num_hidden']
    lr = args['lr']
    batch_size, in_units = 32, 8
    
    class Net(nn.Module):
        def __init__(self, in_units, num_hidden):
            super().__init__()
            self.linear1 = nn.Linear(in_units, num_hidden)
            self.out_layer = nn.Linear(num_hidden, 1)

        def forward(self, x):
            return self.out_layer(nn.ReLU()(self.linear1(x)))
    
    net = Net(in_units=in_units, num_hidden=num_hidden)
    
#     net.cuda()
    net.train()
    optimizer = th.optim.Adam(net.parameters(), lr=lr, amsgrad=True)
    for i in range(10):
        optimizer.zero_grad()
        random_data_in = th.normal(0, 1, (batch_size, in_units))#.cuda()
        random_data_out = 5 * random_data_in ** 2 + 1
        pred_out = net(random_data_in)
        loss = ((random_data_out - pred_out)**2).mean()
        loss.backward()
        with th.no_grad():
#             print(a_gpu)
            print(loss.cpu().detach().numpy())
            reporter(mean_loss=loss.cpu().detach().numpy().item())
        optimizer.step()



scheduler = ag.scheduler.FIFOScheduler(train_nn,
                                   resource={'num_cpus': 4, 'num_gpus': 0},
                                   num_trials=20,
                                   reward_attr='mean_loss',
                                   time_attr='iteration')
scheduler.run()
scheduler.join_jobs()
```
(Have to remove cuda parts since I don't have GPU on my test box)